Electrodialysis_1D documentation

lbibl opened the pull request:
## Checks 1 todo in #489 


## Summary/Motivation:
This PR presents the documentation for the recently merged Electrodialysis_1D unit model. 

## Changes proposed in this PR:
-Minor changes on the Electrodialysis_0D documentation are carried along.


### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

lbibl committed:
1D documentation init

lbibl committed:
merge from main

lbibl committed:
completed draft

lbibl committed:
add index

lbibl committed:
debug

aladshaw3 committed:
Merge branch 'main' into ed1d_doc

aladshaw3 commented on docs/technical_reference/unit_models/electrodialysis_1D.rst:
```diff
@@ -0,0 +1,237 @@
+        }
+
+This model, by default, uses H\ :sub:`2`\ O  as the solvent of the feed solution.
+
```
```suggestion

Information regarding the property package this unit model relies on can be found here: 

:mod:`watertap.property_models.ion_DSPMDE_prop_pack`
```

lbibl replied:
Accepted.

aladshaw3 reviewed (changes requested):
Looks good so far. 

I suggest adding something like I mention below (a link to some form of documentation for the property package). This should probably also be added to the `electrodialysis_0D.rst` doc page as well. 

There currently is no 'technical reference' for the ion property package, but we should at least link to the API docs so a user will have some idea of how that property package is set up.

lbibl committed:
add prop pack reference

lbibl committed:
merge from remote

aladshaw3 reviewed (approved):
LGTM

adam-a-a committed:
subtle cleanup

adam-a-a commented on docs/technical_reference/unit_models/electrodialysis_1D.rst:
```diff
@@ -0,0 +1,241 @@
+   :header: "Symbol", "Description", "Unit"
+   :widths: 10, 20, 10
+
+   "**Variables and Parameters**"
```
It would be good to have Parameters and Variables separate so that we could easily distinguish between the two.

adam-a-a reviewed (changes requested):
To be honest, I haven't looked at the 0D doc yet, but I have to say, this documentation is phenomenal! Extremely well done! I only did some touching up on very minor typos and committed them directly to save you time (please review to make sure you're happy with those). 

The only suggested change that I'd like to see is the separation of the Vars and Parameters. Then, I think it's good to go.

adam-a-a commented on docs/technical_reference/unit_models/electrodialysis_1D.rst:
```diff
@@ -0,0 +1,241 @@
+* Steady state: all variables are independent of time.
+* Co-current flow operation. 
+* Electrical current is operated below the limiting current. 
+* Ideality assumptions: activity, osmotic, and van't Hoff coefficients are set at one.
```
Is the ideality assumption necessarily applicable? The ion prop model has the Davies model, so at the least, activity can deviate from ideality assumptions. Is there something in the ED unit model that would need to be incorporated, beyond what the ion prop model provides (in this case, just referring to the activity)? If not, and the ED model can technically utilize activities calculated by the Davies model (or might be that you'd need to apply activity coefficients--forgot how it was implemented in prop model), then you can boast that the ED model _can _ account for nonideality in terms of activities. My presumption is that you only tested the model with activity coefficients = 1 but the model would work if provided non-ideal activity coefficients. Is that the case?

lbibl replied:
I didn't investigate the activity calculation in the prop model-  is activity defined separately from conc or absorbed into it?    I think for solutions ideality generally suggests particles are point-like and short-range molecular forces are neglected.  I cannot fully specify all non-ideal elements in this scenario at this point, but I am sure it's more than replacing conc by activity.  Things like the charge correlation force gonna be present when salinity is high.  I would keep this statement since it is given in most of the modeling papers.

aladshaw3 committed:
Merge branch 'main' into ed1d_doc

aladshaw3 committed:
Merge branch 'main' into ed1d_doc

lbibl committed:
addresses comments from @adma-a-a

lbibl committed:
update from remote

lbibl commented:
> To be honest, I haven't looked at the 0D doc yet, but I have to say, this documentation is phenomenal! Extremely well done! I only did some touching up on very minor typos and committed them directly to save you time (please review to make sure you're happy with those).
> 
> The only suggested change that I'd like to see is the separation of the Vars and Parameters. Then, I think it's good to go.

Thank you Adam! I tried to make it readable :)  I addressed all your comments (good ones!).  For ideality I would keep the writing as is as the model hasn't touched any non-ideality stuff and we can come back to it in later refinements.   Austin put a hold on checking the files and I think it'll be good to go after that.

adam-a-a commented on docs/technical_reference/unit_models/electrodialysis_1D.rst:
```diff
@@ -0,0 +1,244 @@
+   ":math:`\kappa`", "Solution conductivity", ":math:`S m^{-1}\ or\  \Omega^{-1} m^{-1}`"
+   ":math:`\eta`", "Current efficiency for desalination", "dimensionless"
+   ":math:`P`", "Power consumption", ":math:`W`"
+   ":math:`P_Q`", "Specific power consumption", ":math:`kW\ h\  m^{-3}`"
```
```suggestion
   ":math:`P_Q`", "Specific energy consumption", ":math:`kW\ h\  m^{-3}`"
```

lbianchi-lbl committed:
Merge branch 'main' into ed1d_doc

bknueven committed:
Merge branch 'main' into ed1d_doc
